# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: parser/htmlparser/tests/reftest/in-style.html
# Commit: 9a8abd87cc79
# Full Hash: 9a8abd87cc7935f29b94248c1a6f8203faa14403
# Author: Henri Sivonen <hsivonen@hsivonen.fi>
# Date: 2021-12-08 21:43:06
# Regressor Bug: 1701828
# File Overlap Count: 1
# Description:
#   Bug 1701828 - meta charset rewrite. r=smaug
#   
#   Implements https://github.com/whatwg/html/issues/6962 . Improves performance
#   when <meta charset> occurs in head but after the first kilobyte and aligns
#   behavior better with WebKit and Blink.
# ==============================================================================

diff --git a/parser/htmlparser/tests/reftest/in-style.html b/parser/htmlparser/tests/reftest/in-style.html
new file mode 100644
--- /dev/null
+++ b/parser/htmlparser/tests/reftest/in-style.html
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
+<head>
+<link rel="mismatch" href="references/in-style-ref.html">
+<style><meta charset="windows-1251"></style>
+</head>
+<body>
+<p>Meta in <code>style</code>.</p>
+<p>Test: ï¿½</p>
+<p>If &#x0436;, meta takes effect</p>
+</body>