# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/libeditor/crashtests/1534394.html
# Commit: 0475e2855e5c
# Full Hash: 0475e2855e5c4b03dfce9da151fd4ac2bb1367c3
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2019-05-23 15:52:15
# Regressor Bug: 1534394
# File Overlap Count: 1
# Description:
#   Bug 1534394 - Make WSRunObject::InsertText() set aPointAfterInsertedString by itself when HTMLEditor::InsertTextWithTransaction() returns error r=m_kato
#   
#   Oddly, `WSRunObject::InsertText()` returns `NS_OK` even when
#   `HTMLEditor::InsertTextWithTransaction()` returns error.  However,
#   it fails if insertion point is not editable like `<noscript>` element.
# ==============================================================================

diff -r 4cd6d838796a -r 0475e2855e5c editor/libeditor/crashtests/1534394.html
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/editor/libeditor/crashtests/1534394.html	Thu May 23 06:06:18 2019 +0000
@@ -0,0 +1,29 @@
+<html>
+<head>
+    <script>
+
+      function start () {
+        const sel = document.getSelection()
+        const range_1 = new Range()
+        const noscript = document.getElementById('id_24')
+        const map = document.getElementById('id_26')
+        sel.selectAllChildren(noscript)
+        const range_2 = range_1.cloneRange()
+        range_2.selectNode(map)
+        const frag = range_2.extractContents()
+        range_2.insertNode(noscript)
+        document.documentElement.contentEditable = true
+        document.execCommand('removeFormat', false,)
+        noscript.contentEditable = false
+        document.execCommand('insertText', false, 'ï¿½\n')
+      }
+
+      window.addEventListener('load', start)
+    </script>
+</head>
+<body>
+<big class="">
+    <map class="" id="id_26">
+        <noscript class="" id="id_24">
+</body>
+</html>